test/SparqlAntlrParserTest.cpp

```diff
@@ -661,3 +661,120 @@ TEST(SparqlParser, Bind) {
     EXPECT_THAT(bindAndText._remainingText, IsEmpty());
   }
 }
+
+TEST(SparqlParser, Integer) {
+  {
+    string input = "1931";
+    ParserAndVisitor p{input};
+
+    unsigned long long result =
+        p.parser.integer()->accept(&p.visitor).as<unsigned long long>();
+    EXPECT_EQ(result, 1931ULL);
+  }
+
+  {
+    string input = "0";
+    ParserAndVisitor p{input};
+
+    unsigned long long result =
+        p.parser.integer()->accept(&p.visitor).as<unsigned long long>();
+    EXPECT_EQ(result, 0ULL);
+  }
+
+  {
+    string input = "18446744073709551615";
+    ParserAndVisitor p{input};
+
+    unsigned long long result =
+        p.parser.integer()->accept(&p.visitor).as<unsigned long long>();
+    EXPECT_EQ(result, 18446744073709551615ULL);
+  }
+
+  {
+    string input = "18446744073709551616";
+    ParserAndVisitor p{input};
+
+    EXPECT_THROW(p.parser.integer()->accept(&p.visitor), SparqlParseException);
+  }
+
+  {
+    string input = "10000000000000000000000000000000000000000";
+    ParserAndVisitor p{input};
+
+    EXPECT_THROW(p.parser.integer()->accept(&p.visitor), SparqlParseException);
+  }
+
+  {
+    string input = "-1";
+    ParserAndVisitor p{input};
+
+    EXPECT_THROW(p.parser.integer()->accept(&p.visitor),
+                 antlr4::ParseCancellationException);
+  }
+}
+
+TEST(SparqlParser, LimitOffsetClause) {
+  {
+    string input = "LIMIT 10";
+
+    sparqlParserHelpers::ResultOfParseAndRemainingText<LimitOffsetClause>
```

Comment:
You can use `auto` here and probably should (the type is very long, very expected from the name of the function, etc.
(Also several times below).

Reply:
And I think in this file we can afford a `using namespace sparqlParserHelpers` so it becomse
`auto limitOffset = parseLimitOffsetClause(....)`